cyberbrain/value_stack.py

```diff
@@ -88,16 +84,98 @@ class Why(enum.Enum):
     SILENCED = 8  # Exception silenced by 'with'
 
 
+class SymbolStackItem:
+    """Class representing an item on the value stack to be tracked by Cyberbrain.
+
+    Attributes:
+        start_lineno: The starting lineno of the item.
+        sources: A list of symbols that are the sources of the item.
+    """
+
+    def __init__(
+        self,
+        start_lineno: int,
+        sources: List[Symbol],
+    ):
+        self.start_lineno = start_lineno
+        self.sources = sources
+
+    def __repr__(self) -> str:
+        return "{" + repr(self.start_lineno) + " " + repr(self.sources) + "}"
+
+    def get_top_source(self) -> Symbol:
```

Comment:
would it be better to write?

@property
def top_source

Reply:
Yes, I'll change it